django/forms/widgets.py

```diff
@@ -353,9 +357,13 @@ def get_context(self, name, value, attrs):
                 # An ID attribute was given. Add a numeric index as a suffix
                 # so that the inputs don't all have the same ID attribute.
                 widget_attrs['id'] = '%s_%s' % (id_, index)
+            if self.unique_names:
+                widget_name = '%s_%s' % (name, index)
+            else:
+                widget_name = name
```

Comment:
IMO we don't need a new parameter, the previous behavior was incorrect. We should use different `name`'s for all widgets (like `MultiWidget`).

\cc @carltongibson

Reply:
Yes, I agree: let's just fix the broken behaviour, adding the release note as per [comment](https://github.com/django/django/pull/12567/files#r394830854). 

There's no (non-array, i.e. `name[]`) use-case where multiple inputs with the same `name` is the right behaviour.

Reply:
Having said that, I'm immediately not even sure that we need the release note: `MultipleHiddenInput` is just broken no? (I guess it doesn't hurt...)